processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndexCursorHolder.java

```diff
@@ -239,4 +247,13 @@ private boolean beyondMaxRowIndex(int rowIndex)
       return rowIndex > maxRowIndex;
     }
   }
+
+  private static Order getTimeOrder(List<OrderBy> ordering)
+  {
+    if (!ordering.isEmpty() && ColumnHolder.TIME_COLUMN_NAME.equals(ordering.get(0).getColumnName())) {
```

Comment:
why the first column?

Reply:
ah this is used for stuff like timeseries and timeboundary queries which count on the segment being ordered first by time